web_src/js/features/repo-settings.js

```diff
@@ -11,18 +12,19 @@ export function initRepoSettingsCollaboration() {
     const $dropdown = $(e);
     const $text = $dropdown.find('> .text');
     $dropdown.dropdown({
-      action(_text, value) {
+      async action(_text, value) {
         const lastValue = $dropdown.attr('data-last-value');
-        $.post($dropdown.attr('data-url'), {
-          _csrf: csrfToken,
-          uid: $dropdown.attr('data-uid'),
-          mode: value,
-        }).fail(() => {
+        try {
+          $dropdown.attr('data-last-value', value);
+          $dropdown.dropdown('hide');
+          const data = new FormData();
+          data.append('uid', $dropdown.attr('data-uid'));
+          data.append('mode', value);
+          await POST($dropdown.attr('data-url'), {data});
+        } catch {
           $text.text('(error)'); // prevent from misleading users when error occurs
           $dropdown.attr('data-last-value', lastValue);
```

Comment:
```suggestion
          $dropdown.dropdown('hide');
          const data = new FormData();
          data.append('uid', $dropdown.attr('data-uid'));
          data.append('mode', value);
          await POST($dropdown.attr('data-url'), {data});
          $dropdown.attr('data-last-value', value);
        } catch {
          $text.text('(error)'); // prevent from misleading users when error occurs
```
These two suggestions should be identical to the previous behavior.